pennylane/devices/device_interface.py

```diff
@@ -0,0 +1,399 @@
+# Copyright 2018-2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This module contains the Abstract Base Class for the next generation of devices.
```

Comment:
Could we avoid naming the file `device_interface` as we already use this term for the machine learning one? Then it becomes complicated to know what we talk about when we use the word interface, the name becomes distilled.

Reply:
How about `device_api` instead?

Reply:
This is actually making me confused as to why we call the machine learning packages interfaces in the first place.

Reply:
We are trying to move away from that :) ...if you check the docs we have carefully changed the wording in the introduction pages to call the packages "autodiff libraries", while "interface" only refers to the boundary elements in PL interacting with the library. But could be clearer, I agree!

Reply:
Yep, we have been moving towards:

- `interface`: refers to the internal source code that interfaces between PL and an ML/autodiff library
- ML library/autodiff library: the library a _user uses with PennyLane_. E.g., we avoid saying 'interfaces' explicitly in the user-facing documentation.

Even so, I would avoid using `interface` internally in other parts of the source code (for the time being), just to avoid confusion.